components/Repo/Add.js

```diff
@@ -55,45 +82,146 @@ const styles = {
   })
 }
 
+const TemplatePicker = compose(
+  withT,
+  withRouter,
+  graphql(getTemplateRepos, {
+    options: () => ({
+      fetchPolicy: 'network-only'
+    }),
+    skip: ({ isTemplate }) => isTemplate
+  })
+)(({ t, data, schema, onChange, isTemplate }) => {
+  const [schemaOptions] = useState(
+    schemaKeys.map(key => ({
+      value: key,
+      text: t(`repo/add/template/${key}`, null, key)
+    }))
+  )
+  const [templateFilter, setTemplateFilter] = useState('')
+  const [template, setTemplate] = useState({
+    value: schema,
+    text: t(`repo/add/template/${schema}`, null, schema)
+  })
+
+  const templateOptions = useMemo(() => {
+    return schemaOptions
+      .concat(
+        (data?.reposSearch?.nodes || []).map(repo => ({
+          value: repo.latestCommit.document.meta.template,
+          text:
+            repo.latestCommit.document.meta.title ||
+            repo.id.split('/')[1].replace('-', ' '),
+          repoId: repo.id,
+          slug: repo.latestCommit.document.meta.slug || repo.id.split('/')[1]
+        }))
+      )
+      .filter(
+        ({ text }) =>
+          !templateFilter ||
+          text.toLowerCase().includes(templateFilter.toLowerCase())
+      )
+  }, [data, templateFilter])
+
+  return isTemplate ? (
+    <Dropdown
+      label='Vorlage'
+      items={schemaOptions}
+      value={schema}
+      onChange={item => {
+        onChange({
+          schema: item.value,
+          templateRepoId: undefined,
+          templatePrefix: undefined
+        })
+      }}
+    />
+  ) : (
+    <Loader
+      error={data.error}
+      loading={data.loading}
+      render={() => (
+        <Autocomplete
+          label='Vorlage'
+          value={template}
+          filter={templateFilter}
+          items={templateOptions}
+          onChange={newTemplate => {
+            setTemplate(newTemplate)
+            setTemplateFilter('')
+            onChange({
+              schema: newTemplate.value,
+              templateRepoId: newTemplate.repoId,
+              templatePrefix: newTemplate.slug
+                ? newTemplate.slug.replace('template-', '') + '-'
```

Comment:
Is this necessary? `getSlug` below (where we added the `template-` prefix) will not actually generate an `meta.slug` (traditionally used for published urls) but actually the github repo slug which becomes part of the repo id.

Reply:
i just dont want to get smth like `template-binswager-blah-blah` when an article is created from a template repo with a prefix `template-`. I need to look again at that part then because I'm not sure what I'm missing.